picky-asn1-x509/src/private_key_info.rs

```diff
@@ -398,22 +442,39 @@ impl<'de> serde::Deserialize<'de> for ECPrivateKey {
                     ));
                 }
 
-                Ok(ECPrivateKey {
+                let private_key = seq_next_element!(seq, OctetStringAsn1, "OctetStringAsn1");
+
+                let mut ec_private_key = ECPrivateKey {
                     version,
-                    private_key: seq_next_element!(seq, OctetStringAsn1, "OctetStringAsn1"),
-                    parameters: seq_next_element!(
-                        seq,
-                        Optional<ExplicitContextTag0<Option<EcParameters>>>,
-                        ECPrivateKey,
-                        "EcParameters"
-                    ),
-                    public_key: seq_next_element!(
-                        seq,
-                        Optional<ExplicitContextTag1<BitStringAsn1>>,
-                        ECPrivateKey,
-                        "BitStringAsn1"
-                    ),
-                })
+                    private_key,
+                    parameters: Optional::default(),
+                    public_key: Optional::default(),
+                };
+
+                let mut last_tag = seq.next_element::<TagPeeker>()?;
```

Comment:
Changed logic to parse via `TagPeeker`, because unconditional `seq_next_element!` could throw error, if next element does not exist at all (Even `Optional<...>` wrapper will not help